stan/math/prim/fun/all.hpp

```diff
@@ -0,0 +1,77 @@
+#ifndef STAN_MATH_PRIM_FUN_ALL_HPP
+#define STAN_MATH_PRIM_FUN_ALL_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/functor/for_each.hpp>
+#include <algorithm>
+
+namespace stan {
+namespace math {
+
+/**
+ * Return true if all values in the input are true.
+ *
+ * Overload for a single boolean input
+ *
+ * @param x boolean input
+ * @return The input unchanged
+ */
+inline bool all(bool x) { return x; }
```

Comment:
(optional) constexpr?

Reply:
Also should this be templated to accept any integral type? Not sure if we would get warnings about downcasting? This applies to all the bool as input functions